src/components/tools/hooks/use-measure-text.ts

```diff
@@ -1,15 +1,20 @@
-import { useCallback, useRef } from "react";
+import { useCallback } from "react";
 
-export const useMeasureText = (font: string) => {
-  const cache = useRef<Record<string, number>>({});
-  const canvas = useRef(document.createElement("canvas"));
-  const context = canvas.current.getContext("2d");
+const defaultFont = "italic 14px Lato, sans-serif";
+const canvas = document.createElement("canvas");
+const cache: Record<string, Record<string, number>> = {};
+
+export const measureText =
+  (text:string, font = defaultFont) => {
+  const context = canvas.getContext("2d");
   context && font && (context.font = font);
+  cache[font] = cache[font] || {};
+  cache[font][text] = cache[font][text] || (context ? Math.ceil(10 * context.measureText(text).width) / 10 : 0);
+  return cache[font][text];
+};
```

Comment:
This is inarguably a more elegant and ultimately more efficient implementation which I considered when doing my previous review. I didn't push for it, however, because it's also a somewhat riskier implementation in that when every client is sharing the same cache there's more potential for a bug to result in cross-contamination. In the previous implementation, the most likely use case would be a single client measuring text in a single context, e.g. the image title where all of this began. If a client is only ever asking for a couple of different strings in the same font there's little that can go wrong. With a single global cache the risk factor is multiplied. That said, having done the work to develop the more elegant and performant implementation, rather than going back we should endeavor to reduce the risks, to which end I would suggest adding some unit tests that validate that the cache is working correctly and that the right value is returned when fonts are changed, etc. They should be straightforward jest tests to write, and I think you can focus on `measureText()` itself, since `useMeasureText()` is such a thin wrapper around `measureText()`.